src/daemons/GraphDaemon.cpp

```diff
@@ -89,15 +89,12 @@ int main(int argc, char *argv[]) {
     }
 
     // Get the IPv4 address the server will listen on
-    std::string localIP;
-    {
-        auto result = NetworkUtils::getIPv4FromDevice(FLAGS_listen_netdev);
-        if (!result.ok()) {
-            LOG(ERROR) << result.status();
-            return EXIT_FAILURE;
-        }
-        localIP = std::move(result).value();
+    if (FLAGS_local_ip.empty()) {
```

Comment:
The  local_ip is not allowed empty, Because it's stored in Metad, it is meaningless when empty.

Reply:
I mean why change the logic of `getIPv4FromDevice` to this new one.

Reply:
The `getIPv4FromDevice` will get the 0.0.0.0 ip.

Reply:
Does this related to session?

Reply:
the session info will save the graph address, but it's not a direct relationship. It is only because local_ip must be set, otherwise the identities of different GraphDs may become indistinguishable on the machine.